### ISSUE TITLE
Support groovy compilation

### ISSUE THREAD

**Issue by BrynCooke** [tbroyer/gradle-apt-plugin#20]
Annotation processing works with Groovy compilation now.
It's be great if this plugin supported this.

**Comment by tbroyer** [comment]
Do you think the plugin should automatically set `compileGroovy.groovyOptions.javaAnnotationProcessing` to `true`? Or only configure the `compileGroovy` and `compileTestGroovy` tasks?

**Comment by BrynCooke** [comment]
I must admit that I am new to Gradle.
Gut feeling from looking at your source is that the compile*Groovy tasks need configuring.
Just to give you some context I noticed this issue when trying to use the plugin on a mixed mode groovy/java project. Without the plugin things actually compiled OK using gradle, but Idea is broken. With the plugin, Idea is OK, but won't compile with gradle.

**Comment by BrynCooke** [comment]
Taking a look at the groovy compiler I am not even sure this is possible.
Groovyc doesn't seem to have a -s option.
http://groovy-lang.org/groovyc.html

**Comment by tbroyer** [comment]
Annotation processing for Groovy involves generating Java "stubs" that are then processed by JavaC.

**Comment by BrynCooke** [comment]
So possibly the -Fflag option could be used?

**Comment by tbroyer** [comment]
Actually, we don't even need to think about it, Gradle (apparently) already handles everything correctly; the only question really is: should `compileGroovy.groovyOptions.javaAnnotationProcessing` be forcefully set to `true` or left to its default value? (I lean towards the latter)

**Comment by BrynCooke** [comment]
For me at least this didn't work.
```
compileGroovy {
        groovyOptions.javaAnnotationProcessing = true
    }
```
produced no generated source code and the compilation failed.

Using:
```
compileGroovy {
        groovyOptions.javaAnnotationProcessing = true
        options.compilerArgs += [
            "-s", "${project.buildDir}/generated/source/apt",
        ]
    }
```
did work, but obviously the source is not actually generated in the correct location (needs to be split in to main and test). Also the target directory needs to be created by hand.

**Comment by tbroyer** [comment]
Should be fixed in e930273ec6bd3fbdca253319ad1289653e89973c but I haven't actually tested IDE integration. Will cut a release shortly.

**Comment by tbroyer** [comment]
v0.6 released: https://plugins.gradle.org/plugin/net.ltgt.apt/0.6
